test/task_test.rb

```diff
@@ -87,7 +87,7 @@
         syskit_configure_and_start(task)
         t0 = Time.now
         pwm_out = expect_execution do
-            syskit_write task.cmd_in_port, effort_command({ a: -16.45 })
+            syskit_write task.cmd_in_port, effort_command({ a: -10.45 })
```

Comment:
I don't agree with these changes, currently inverting the command result in inverting the thruster torque to pwm table. Which in my opinion is the correct interpretation. For instance, now the table maps from -10 torque to 1100 pwm, to 16 torque for 1900 pwm. And for the inverted *Helice* It *should* be 10 torque for 1100 pwm, and -16 torque for 1900 pwm. As inverting the helice IMO *just* inverts the direction of the force.

What you are doing here is masquerading the problem, when receiving a -10 torque, you shouldn't be saturating for an inverted helice, as an inverted helice can output up to -16 torque. So you are incorrectly transforming an -10 torque command into -16 torque.

The real problem that should fix what you observed is changing the Thruster allocation task and change the *force limits* for the *inverted* thrusters. e.g:

aft_starboard_max_force:
  positive: 16.44
  negative: -10.21
  
 Should be 
 
aft_starboard_max_force:
  positive: 10.21
  negative: -16.44

As this is the correct physical interpretation of the thruster's torque(as it has an inverted helice).

Reply:
The current interpretation is wrong. Here's why:

The inversion that happens here is because of the counter-clockwise propeller. Having a CCW thruster means that we have to invert the direction of rotation to get the same flow.

Meaning that a PWM < 1500 with a CCW propeller is *as efficient as* a PWM > 1500 with a CW propeller.

What the current implementation does is assume that the effort is reversed because of CW/CCW. It is not. What you need to do for a CCW propeller is invert the PWM to get the same effort.